packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.m

```diff
@@ -264,7 +264,27 @@ - (void)setTextContentType:(NSString *)type
       @"password" : UITextContentTypePassword,
       @"newPassword" : UITextContentTypeNewPassword,
       @"oneTimeCode" : UITextContentTypeOneTimeCode,
-    };
+    } mutableCopy];
+
+    if (@available(iOS 17.0, *)) {
```

Comment:
let's add compile time check here too

__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_17_0

Reply:
Added, which has resolved the test issues